src/codelabs/async-await/index.md

```diff
@@ -0,0 +1,346 @@
+result of type T. For example,  
+`Future<String>` is the type signature for a future object that produces a string
+result. If a future produces a result that isn't a usable value, then the
+future's type is `Future<void>`. 
+
+#### 2. Error
+
+If the asynchronous operations being performed by the 
+function throw an error, the future completes and returns an error. 
+
+To see this lifecycle at work consider what happens after invoking the 
+`getUserOrder` function:
+
+
+<iframe style="margin-top: 25px; margin-bottom: 25px" frameborder="no" height="450" src="https://dartpad.dartlang.org/experimental/embed-new.html?id=57e6085344cbd1719ed42b32f8ad1bce" width="100%"></iframe>
+
+
+After 1 second has passed, notice that the future is in an uncompleted state - you can't access the
+desired result yet:
+
+`     Your order is: Instance of '_Future<String>'`
+
+After 4 seconds have passed, the future completes and produces the desired
+result:
+
+`Your order is: 'Large Latte'`
+
+How do you access this result that completes 4 seconds later? In the next section
+you will learn about using  the `async` and `await` keywords to handle the results
+of asynchronous functions. 
+
+{{ site.alert.secondary }}
+Quick Review:
+
+* In Dart, a
+[Future]({{site.dart_api}}/Future-class.html)
+produces a result of type `<T>`. For example,
+[Future]({{site.dart_api}}/Future-class.html)`<String>`
+is the type signature for a future object that produces a string result. If a
+future produces a result that isn't a usable value, then the future's type is
+`Future<void>`.
+* A future can be in one of two states: uncompleted, or completed. 
+* A completed future can either (1) complete with a result or (2) complete with
+an error.
+* When you invoke a function that returns a future, two things happen:
+
+1. The function queues up work to be done and returns an uncompleted future.
+2. Later, when the operation is finished, the future completes with a value or
+with an error.
+{{ site.alert.end }}
+
+
+## Working with futures in Dart: async and await
+
+The `async` and `await` keywords provide a declarative way to: Define 
+asynchronous functions and access their results *after they have completed.*
+
+To define an `async` function:
+
+1. Add the `async` keyword after the function's parameters but before the function's body:  
+
+    ```
+    createOrderMessage () async {
+    ```
+
+2. Update the function's type signature to return a future. The following  
+example defines an `async` function that completes with a string.  
+
+    ```
+    Future<String> createOrderMessage() async {
+    ```
+
+{{ site.alert.info }}
+ You can 
+ [use Future\<void\>]({{url}}/guides/language/effective-dart/design#do-use-futurevoid-as-the-return-type-of-asynchronous-members-that-do-not-produce-values)
+ as the return type for asynchronous functions that don't return usable values.
+{{ site.alert.end }}
+
+Now that you've declared an async function, it returns a future object. Use the
```

Comment:
-reading this sentence makes me think I've been following a how-to manual.

Reply:
That's probably a good thing -- this is the end of a section describing how to use the async and await keywords. Let me know if you have additional thoughts on this.